x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curations_logic.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter, mockHttpValues, mockFlashMessageHelpers } from '../../../__mocks__';
+import '../../__mocks__/engine_logic.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { DEFAULT_META } from '../../../shared/constants';
+
+import { CurationsLogic } from './';
+
+describe('CurationsLogic', () => {
+  const { mount } = new LogicMounter(CurationsLogic);
+  const { http } = mockHttpValues;
+  const { clearFlashMessages, setSuccessMessage, flashAPIErrors } = mockFlashMessageHelpers;
+
+  const MOCK_CURATIONS_RESPONSE = {
+    meta: {
+      page: {
+        current: 1,
+        size: 10,
+        total_results: 1,
+        total_pages: 1,
+      },
+    },
+    results: [
+      {
+        id: 'some-curation-id',
+        last_updated: 'January 1, 1970 at 12:00PM',
+        queries: ['some query'],
+        promoted: [],
+        hidden: [],
+        organic: [],
+      },
+    ],
+  };
+
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    curations: [],
+    meta: DEFAULT_META,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(CurationsLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('onCurationsLoad', () => {
+      it('should set curations and meta state, & dataLoading to false', () => {
+        mount();
+
+        CurationsLogic.actions.onCurationsLoad(MOCK_CURATIONS_RESPONSE);
+
+        expect(CurationsLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          curations: MOCK_CURATIONS_RESPONSE.results,
+          meta: MOCK_CURATIONS_RESPONSE.meta,
+          dataLoading: false,
+        });
+      });
+    });
+
+    describe('onPaginate', () => {
+      it('should set meta.page.current state', () => {
+        mount();
+
+        CurationsLogic.actions.onPaginate(3);
+
+        expect(CurationsLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          meta: { page: { ...DEFAULT_VALUES.meta.page, current: 3 } },
```

Comment:
Don't change this as I'm sure this was just migrated as is. This is just a comment for us to think about in the future.

If we have state like this that needs to be updated regularly, it can be much nicer to normalize state a bit so that we don't have to reach deeply inside of an object like this to do an update. So in this case we'd do something like pull `currentPage` into a top level state property.

Reply:
This was a pretty big pain with the Relevance Tuning UI work, as I think you noticed in all of the actions that were updating nested boost objects.

Reply:
++, I have a helper for this coming in the next upcoming pagination PR refactor. It only applies specifically to the `meta` nested obj however so it wouldn't be super helpful for relevance tuning. Generally I'm a +1 for pulling out to a top-level var, but in the specific case of `meta` I think there might be an exception to be made there (since the format is coming from our API and doesn't really change). Happy to discuss more in that PR